SpatialGDK/Source/SpatialGDK/Private/Interop/SpatialReceiver.cpp

```diff
@@ -137,38 +142,61 @@ void USpatialReceiver::OnAddComponent(Worker_AddComponentOp& Op)
 		return;
 	}
 
-	// If a client gains ownership over something it had already checked out, it will
-	// add component interest on the owner only data components, which will trigger an
-	// AddComponentOp, but it is not guaranteed to be inside a critical section.
-	if (!NetDriver->IsServer())
+	if (bInCriticalSection)
 	{
-		if (USpatialActorChannel* Channel = NetDriver->GetActorChannelByEntityId(Op.entity_id))
-		{
-			if (ClassInfoManager->GetCategoryByComponentId(Op.data.component_id) == SCHEMA_OwnerOnly)
-			{
-				// We received owner only data, and we have the entity checked out already,
-				// so this happened as a result of adding component interest. Apply the data
-				// immediately instead of queuing it up (since there will be no AddEntityOp).
-				ApplyComponentData(Op.entity_id, Op.data, Channel);
-				return;
-			}
-		}
+		PendingAddComponents.Emplace(Op.entity_id, Op.data.component_id, MakeUnique<DynamicComponent>(Op.data));
```

Comment:
Why make unique? This is already an RAII wrapper.

Reply:
You're right, though this is how the code already was, using the FPendingAddComponents wrapper.